core/templates/dev/head/domain/collection/CollectionValidationService.js

```diff
@@ -72,6 +72,28 @@ oppia.factory('CollectionValidationService', [
       });
     };
 
+    // Validates that the tags for the collection are in the proper format,
+    // returns true if all tags are in the correct format.
+    var validateTagFormat = function(tags) {
+      // Check to ensure that all tags follow the format specified in
+      // TAG_REGEX.
+      for (var i = 0; i < tags.length; i++) {
```

Comment:
Oh, sorry I missed this earlier, I kind of meant to do something like
```
var tagRegex = new RegExp(GLOBALS.TAG_REGEX);
return tags.every(function(tag) {
  return tag.match(tagRegex);
});
```
This way you also get the `tagRegex` construction out of the loop.